File: backend/CineNiche/Data/Movie.cs
using System.ComponentModel.DataAnnotations;

public class Movie
{
    [Key]
    public string show_id { get; set; }
    public string? title { get; set; }
    public string? director { get; set; }
    public string? cast { get; set; }
    public string? country { get; set; }
    public int? release_year { get; set; }
    public string? rating { get; set; }
    public string? duration { get; set; }
    public string? description { get; set; }
}

File: frontend/src/types/Movie.ts
export interface MovieCardProps {
  show_id: string;
  title: string;
  description?: string;
  releaseYear?: number;
  duration?: string;
  genres?: string[];
  director?: string;
  cast?: string;
  rating?: number;
  posterUrl?: string;
  recommended?: Array<{
    show_id: string;
    title: string;
    posterUrl?: string;
  }>;
  userRatings?: number[]; // Added userRatings property
}